bigquery/snippets/snippet.go

```diff
@@ -242,7 +317,67 @@ func asyncQuery(client *bigquery.Client, datasetID, tableID string) error {
 		}
 		fmt.Println(row)
 	}
-	// [END bigquery_async_query]
+	// [END bigquery_query]
+	return nil
+}
+
+func printTableMetadataSimple(client *bigquery.Client, datasetID, tableID string) error {
+	ctx := context.Background()
+	// [START bigquery_get_table]
+	meta, err := client.Dataset(datasetID).Table(tableID).Metadata(ctx)
+	if err != nil {
+		return err
+	}
+	// Print information about the table
+	fmt.Printf("Schema: %+v\n", meta.Schema)
+	fmt.Printf("Description: %s\n", meta.Description)
+	fmt.Printf("Row in managed storage: %d\n", meta.NumRows)
+	// [END bigquery_get_table]
+	return nil
+}
+func printTableMetadataExtended(client *bigquery.Client, datasetID, tableID string) error {
+	ctx := context.Background()
+	// Demonstrates some of the information available in the metadata for a table.
+	meta, err := client.Dataset(datasetID).Table(tableID).Metadata(ctx)
+	if err != nil {
+		return err
+	}
+	// Print information about the table
+	fmt.Printf("Table ID: %s\n", tableID)
+	if len(meta.Name) > 0 {
+		fmt.Printf("Has Friendly Name: %s", meta.Name)
+	}
+
+	if meta.Type == bigquery.ViewTable {
+		// Table is a logical view, rather than a table with backing storage
+		fmt.Println("Table is a logical view")
+	} else {
```

Comment:
Should this be `else if` ? It reads a bit strange right now nested another layer.

Is the reason for current structure that you can't print schema for view table?

Reply:
Pulled this from the snippets for now, may introduce it later as a more extended example.  Logic is a bit odd because some of the messages bridge types (e.g. I wanted to show schema for native/external tables, but skip schema for view since stale schema on logical views tends to trip up people).